packages/relay-compiler/bin/RelayCompilerBin.js

```diff
@@ -163,9 +165,10 @@ Ensure that one such file exists in ${srcDir} or its parents.
   };
   const writerConfigs = {
     js: {
-      getWriter: getRelayFileWriter(srcDir),
+      getWriter: getRelayFileWriter(srcDir, options.persist),
       isGeneratedFile: (filePath: string) =>
-        filePath.endsWith('.js') && filePath.includes('__generated__'),
+        (filePath.endsWith('.js') || filePath.endsWith('.queryMap.json')) &&
```

Comment:
Wondering if it would make sense to follow the `*.graphql.extname` filename pattern used by the other artifacts? Because as shown in the patch in [this comment](https://github.com/facebook/relay/pull/2293#issuecomment-369036035) it wouldn’t be unthinkable to simplify artefact detection somewhat by not matching a specific extension.

Reply:
This is an interesting one. I'm not sure the query map file qualifies as a `graphql` file. It's a plain json file and naming it `graphql` could be misleading.

Reply:
Agreed that it’s a tricky question. The JS artefacts also aren’t strictly GraphQL files but also derivatives, so figured that applies to these as well.

(Just to be clear, I meant something like `Foo.graphql.json`)

Reply:
That's a good point actually I never thought about that. I'll go ahead with the rename. Thanks for the response!

Reply:
@alloy
>Wondering if it would make sense to follow the *.graphql.extname filename pattern used by the other artifacts? Because as shown in the patch in this comment it wouldn’t be unthinkable to simplify artefact detection somewhat by not matching a specific extension.

Done

Reply:
Ace 👍

Reply:
@yusinto Hrmm, I’ve been trying to integrate this PR into the build we use at Artsy and I’m finding that in our case the (single query map) json file gets loaded at runtime _before_ the source artefact file (a `.tsx` file in our case). Have you been able to use this exact code in production?

Changing these filenames back to `*.queryMap.json` makes it work well.